src/utils.rs

```diff
@@ -50,3 +53,120 @@ pub fn get_screen_size() -> Size {
+///
+/// This function looks up the property `window[name]` on the global window, checks that it is a JavaScript
+/// function, and then calls it using `Function.prototype.apply`.
+///
+/// # Type Parameters
+///
+/// * `S: AsRef<str>` – A type that can be converted to a string slice; used for the name of the function.
+/// * `T: Into<JsValue>` – A type that can be converted into a `JsValue`, representing the `this` binding.
+/// * `I: IntoIterator` – An iterable collection of function arguments.
+/// * `I::Item: Into<JsValue>` – Each argument can be converted into a `JsValue`.
+///
+/// # Parameters
+///
+/// * `name` - The name of the JavaScript function (as a property on the global `window` object).
+/// * `this` - The value to use as the `this` binding during the function invocation.
+/// * `args` - An iterable list of arguments to pass to the function.
+///
+/// # Returns
+///
+/// * `Ok(JsValue)` with the result of the function call if successful.
+/// * `Err(Error)` if the window is not available, if the function is not found,
+///   or if the function call fails (e.g. due to a JavaScript exception).
+///
+/// # Examples
+///
+/// Calling a global JS function with a custom context:
+///
+/// ```no_run
+/// # use wasm_bindgen::JsValue;
+/// # use ratzilla::utils::call_js_function_with_context;
+/// # use ratzilla::error::Error;
+/// # fn example() -> Result<(), Error> {
+/// // Suppose `myObj` is a JS object you want to be the `this` value.
+/// let my_obj = JsValue::from(js_sys::Object::new());
+/// let result = call_js_function_with_context(
+///     "myJsFunction",  // JavaScript global function name
+///     my_obj,          // custom `this` context
+///     [JsValue::from(42), JsValue::from("hello")] // arguments
+/// )?;
+/// # Ok(())
+/// # }
+/// ```
+///
+/// Calling a function without caring about `this`:
+///
+/// ```no_run
+/// # use ratzilla::utils::call_js_function_with_context;
+/// # use ratzilla::error::Error;
+/// # use wasm_bindgen::JsValue;
+/// # fn example() -> Result<(), Error> {
+/// // This will set `this` to `null` which in non-strict mode defaults to the global object.
+/// let result = call_js_function_with_context("alert", JsValue::NULL, ["Hello from Rust"])?;
+/// # Ok(())
+/// # }
+/// ```
+pub fn call_js_function_with_context<S, I, T>(name: S, this: T, args: I) -> Result<JsValue, Error>
+where
+    S: AsRef<str>,
+    T: Into<JsValue>,
+    I: IntoIterator,
+    I::Item: Into<JsValue>,
+{
+    let window = web_sys::window().ok_or(Error::UnableToRetrieveWindow)?;
+    let func_val = Reflect::get(&window, &JsValue::from_str(name.as_ref())).map_err(Error::from)?;
+    let func = func_val.dyn_into::<Function>().map_err(Error::from)?;
+    let param_array: Array = args.into_iter().map(Into::into).collect();
+    let ctx = this.into();
+    let result = func.apply(&ctx, &param_array).map_err(Error::from)?;
```

Comment:
The error conversions can be implicit:

```suggestion
    let window = web_sys::window().ok_or(Error::UnableToRetrieveWindow)?;
    let func_val = Reflect::get(&window, &JsValue::from_str(name.as_ref()))?;
    let func = func_val.dyn_into::<Function>()?;
    let param_array: Array = args.into_iter().map(Into::into).collect();
    let ctx = this.into();
    let result = func.apply(&ctx, &param_array)?;
```